jobinfo_test.go

```diff
@@ -282,8 +284,902 @@ func TestJobInfo_GetXML(t *testing.T) {
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>5</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>10</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>15</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>20</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>25</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>30</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>35</tasks>
+		  </job_list>
+		  <job_list state="running">
+			<JB_job_number>1006</JB_job_number>
+			<JAT_prio>0.55500</JAT_prio>
+			<JB_name>task_array.sh</JB_name>
+			<JB_owner>darrellb</JB_owner>
+			<state>r</state>
+			<JAT_start_time>2019-11-15T11:31:47</JAT_start_time>
+			<slots>1</slots>
+			<tasks>40</tasks>
+		  </job_list>
+		</Queue-List>
+	  </queue_info>
+	  <job_info>
+		<job_list state="pending">
+		  <JB_job_number>1006</JB_job_number>
+		  <JAT_prio>0.55500</JAT_prio>
+		  <JB_name>task_array.sh</JB_name>
+		  <JB_owner>darrellb</JB_owner>
+		  <state>qw</state>
+		  <JB_submission_time>2019-11-15T11:31:34</JB_submission_time>
+		  <slots>1</slots>
+		  <tasks>41-150:1</tasks>
+		</job_list>
+	  </job_info>
+	</job_info>`
+
+	ji, err := NewJobInfo(input)
+
+	assert.Nil(t, err)
+
+	assert.NotEmpty(t, ji.PendingJobs.JobList)
+	assert.Len(t, ji.PendingJobs.JobList, 110)
+	assert.Equal(t, int64(41), ji.PendingJobs.JobList[0].Tasks.TaskID)
```

Comment:
I'd like to add a full assert.Equal for the entire Job given we know what the structure would look like. This wouldn't catch if things screwed up like the slots/state etc. Would we do 3 different checks the first element in the pending array, one in the middle, and the last.

Likewise, a full equal check for one of the running jobs

Reply:
Went through and picked the first, middle and last of the task range and verified based on discrete objects they were equal.